eng/Packages.Data.props

```diff
@@ -163,6 +163,7 @@
     <!-- TODO: Make sure this package is arch-board approved -->
     <PackageReference Update="Microsoft.IdentityModel.Protocols.OpenIdConnect" Version="6.35.0" />
     <PackageReference Update="Microsoft.IdentityModel.Tokens" Version="6.35.0" />
+    <PackageReference Update="OpenAI" Version="2.0.0-beta.7" />
```

Comment:
I feel like we probably want to move this into a conditional that allows it ONLY for the Azure OpenAI package.   I don't think we intend to let this be a dependency used by other packages in the repro.  

[Example from L169](https://github.com/Azure/azure-sdk-for-net/blob/main/eng/Packages.Data.props#L169):

```xml
<ItemGroup Condition="$(MSBuildProjectName.StartsWith('Azure.Monitor.OpenTelemetry'))">
```

Reply:
Would it not be simpler in that case to simply not have that dependency listed here in the general file in that case?

Reply:
We generally prefer not allowing libraries to include long-term `VersionOverride` attributes, so we'll want to keep the version centralized with the scope.